tutorials/solids/linearElasticity/flatEndedRigidIndenter/Allrun

```diff
@@ -13,20 +13,38 @@ solids4Foam::convertCaseFormat .
 solids4Foam::runApplication blockMesh
 
 # Mirror mesh
-solids4Foam::runApplication mirrorMesh 
-mv 0/polyMesh/* constant/polyMesh/
-rm -rf 0/polyMesh
+if [[ "${WM_PROJECT}" == "foam" ]]
+then
+    solids4Foam::runApplication mirrorMesh
+    mv 0/polyMesh/* constant/polyMesh/
+    rm -rf 0/polyMesh
+else
+    solids4Foam::runApplication mirrorMesh -overwrite 
+fi
 
 # Run solver
 solids4Foam::runApplication solids4Foam
 
 # Create plot
-sample -latestTime > log.sample
+if [[ "${WM_PROJECT}" == "foam" ]]
+then
+    solids4Foam::runApplication sample -latestTime
+else
+    solids4Foam::runApplication postProcess -func sampleDict.v2012 -latestTime
+fi
 
 if command -v gnuplot &> /dev/null
 then
-    echo "Generating plot using gnuplot"
-    gnuplot plot.gnuplot &> /dev/null
+    if [[ -f "./postProcessing/surfaces/1/sigma_surface.raw" ]]
+    then
+        echo "Generating plot using gnuplot"
+        gnuplot plot.gnuplot &> /dev/null
+    else
+        echo "Generating plot using gnuplot"
+        sed -i  "s|path = .*|path = \"postProcessing/sampleDict.v2012/1/sigma_surface.raw\"|g" plot.gnuplot
```

Comment:
Hi @iBatistic,

@cloner0110 and I were looking for a general solution for fixing these gnuplot paths; we have initially proposed a fix in the  `solids4Foam::convertCaseFormat ` function: see https://github.com/solids4foam/solids4foam/pull/58. I think we can edit the current file to work with that fix (or update that fix as needed).

Philip

Reply:
For now, Ivan, your fix works, so it makes sense to merge it. Amirhossein or you may like to make a separate request to make this gnuplot versions behaviour consistent.

Reply:
Ok, so you propose to update the `solids4foam::convertCaseFormat` function and remove this sed command from Allrun?

Reply:
Note that I pushed the code for of version check and now it ended up in this PR

Reply:
> Ok, so you propose to update the `solids4foam::convertCaseFormat` function and remove this sed command from Allrun?

Yep, I think that will be the most general solution; that way, we can follow a similar convention every time gnuplot is used. What do you think?

... although it could be great if this and other functions of solids4foam were actually documented on the website; that is another day's work!

Reply:
> Yep, I think that will be the most general solution; that way, we can follow a similar convention every time gnuplot is used. What do you think?

Of course; I don't know a better solution.

> .. although it could be great if this and other functions of solids4foam were actually documented on the website; that is another day's work!

Do you mean primarily `solids4foam::convertCaseFormat` and `solids4foam::convertCaseFormatFoamExtend`? I guess documenting this can be very useful.  If you want I can try make some documentation for these functions.

Reply:
Yep, those are the functions that I was referring to.
Also, as the tutorials guide on the website explands (thanks greatly to you!), it becomes clearer that the other parts of solids4foam also need documentation. On the website, I am thinking we could broadly follow the same strucutre as the solids4foam directory tree, e.g.: something like
```
Tutorials Guide
-> a few tutorials as a starting point (I think it is good to keep these)
-> more tutorials
    -> solids, fluids, fluidSolidInteraction: -> add a README for every tutorial

Documentation
-> solver -> short explanation of the solids4Foam solver code
-> utilities -> short explanation of each utility that comes with soldis4foam, i.e. at least explain what it does and what inputs it takes
-> scripts -> solids4FoamScripts here and the other scripts
-> src
    -> solidModels, fluidModels, fluidSolidInterfaces, boundary conditions, mechanical laws : -> describe main important features here and some coding-related parts, e.g. how to add a new mechanical law or boundary condition
-> Third party: explain what this is, and why it is needed: -> actually, we may be able to get rid of Eigen entirely as the ESI version now has some nice dense matrix functionality (although maintaining all versions may be a challenge).
```
... my initial thoughts! As always, I am open to suggestions.

Reply:
I also think that the best option is to follow s4f directory tree structure. This should be the best option for a new user. Now that I understand your plan with documentation, I will try to make some parts :)